# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformFontList.cpp
# Commit: 83815c4cf97f
# Full Hash: 83815c4cf97ff21836996defd83442e843d23e4f
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-02-23 21:42:28
# Description:
#   Bug 1616798 - Ensure SetCharMapRunnable holds a reference to the charmap. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D63461
# ==============================================================================

diff -r b4e30e9317ad -r 83815c4cf97f gfx/thebes/gfxPlatformFontList.cpp
--- a/gfx/thebes/gfxPlatformFontList.cpp	Sun Feb 23 12:06:40 2020 +0000
+++ b/gfx/thebes/gfxPlatformFontList.cpp	Sun Feb 23 12:15:00 2020 +0000
@@ -2265,6 +2265,7 @@
 void gfxPlatformFontList::SetCharacterMap(uint32_t aGeneration,
                                           const fontlist::Pointer& aFacePtr,
                                           const gfxSparseBitSet& aMap) {
+  MOZ_ASSERT(XRE_IsParentProcess());
   auto list = SharedFontList();
   MOZ_ASSERT(list);
   if (!list) {
@@ -2275,7 +2276,7 @@
   }
   fontlist::Face* face = static_cast<fontlist::Face*>(aFacePtr.ToPtr(list));
   if (face) {
-    face->SetCharacterMap(list, &aMap);
+    face->mCharacterMap = GetShmemCharMap(&aMap);
   }
 }
 
